xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -1103,7 +1103,13 @@ bool CGUIWindowVideoBase::OnPlayAndQueueMedia(const CFileItemPtr &item, std::str
   if (iPlaylist != PLAYLIST_NONE && CServiceBroker::GetPlaylistPlayer().IsShuffled(iPlaylist))
      CServiceBroker::GetPlaylistPlayer().SetShuffle(iPlaylist, false);
 
-  CFileItemPtr movieItem(new CFileItem(*item));
+  const CFileItemPtr movieItem = std::make_shared<CFileItem>(*item);
+  if (movieItem->IsVideoDb())
+  {
+    movieItem->SetPath(item->GetVideoInfoTag()->m_strFileNameAndPath);
```

Comment:
I think you can achieve the same behaviour without having to use SetPath by calling IsVideoDB on item. Then construct new FileItem with desired path.

Reply:
Do you mean

```c++
std::shared_ptr<CFileItem> movieItem;
if (item->IsVideoDb())
 movieItem = std::make_shared<CFileItem>(item->GetVideoInfoTag()->m_strFileNameAndPath);
else
 movieItem = std::make_shared<CFileItem>(*item);
```

I fear that the newly item constructed with just the file path will miss some information the original item contains and that is needed by the code it will be passed to. This change is far more risky than what I suggest.

Reply:
SetPath is not supposed to be used and should disappear in the future, see comment in code. Path is the id of a FileItem object. If you change the path you can't identify the item anymore. Saying this I am not sure if this PR is correct. One potential issue is that GUI won't be able to highlight currently playing item.

Reply:
> SetPath is not supposed to be used and should disappear in the future

 Agreed, but this is a looooooooooong way to go. We currently do this all over the place.

 My point is, that at current stage and only at that stage (we are nearing RC for v18) my appraoch is okay as it 

a) works (as it just copies what the corresponding alternative code path does for ages)
b) is minimalistic, so risk of regression is low
c) we urgently need a fix, because very basic functionality is currently broken. This is a v18 relase blocker, imo.
d) nobody came up with an alternative approach, in fact nobody took care about this problem that was reported weeks ago already

Reply:
> How should GUI know which listItem to highlight if you change path here?

 Because there are other ugly "hacks" (formerly maybe official features) in place for this. Last weekend, I have learned about window property "original_listitem_url", which is used to store the original item path whenever it gets overridden. Code has to know wheter it has to use GetPath() or GetProperty("original_listitem_url"). This is kinda item dyn url version 1... 

 Example: 

```c++
CFileItem item(*pItem);
  if (pItem->IsVideoDb())
  {
    item.SetPath(pItem->GetVideoInfoTag()->m_strFileNameAndPath);
    item.SetProperty("original_listitem_url", pItem->GetPath());
}
```
> Addons may listen to OnPlaybackStarted and will get wrong item path now, etc ...

 If they do so they would run into trouble even today, as the code path taken if auto-play-next-item is not active behaves for years(!) exactly like the code path taken if auto-play-next-item is active will behave after my change. With respect to that, I consider this even an improvement, because it is at least consistent now.

Reply:
as ksooo mentioned, there are many places throughout the codebase that create copies of the actual file item, change it's path and then pass it to the UI (like the logic that opens the video info dialog IIRC).
The current approach IMO looks like a valid fix for v18 since `CGUIWindowVideoBase::OnPlayMedia()` a couple lines above does exactly the same thing.